### PR TITLE
Default last used

### PR DESCRIPTION
**Pull request by JustinFreitas**
This pull request should be included if you want the default values in the clock adjuster (i.e. add hour, add minute, add day, add month, add year) to be populated from THE LAST USED/DOUBLE-CLICKED VALUE.

### CODE REVIEW

**Comment by bmos** [comment]
This is great! Thank you so much Justin.

File: scripts/manager_time.lua [hunk 1]
```diff
@@ -18,23 +18,13 @@ function onInit()
...
-	-- Debug.console("setStartTime called; " .. sFirst .."");
-	local nodeActor = rActor;
-	nStartTime = getCurrentDateinMinutes(rActor);
-	-- Debug.console("setStartTime; nStartTime =", nStartTime);
+function setStartTime(nodeActor)
+	local nStartTime = getCurrentDateinMinutes();
 	DB.setValue(nodeActor, "starttime", "number", nStartTime);
-	Debug.console("setStartTime", rActor, sFirst, nStartTime, DB.getValue(nodeActor, "starttime"));
-	-- Debug.console("setStartTime; DB.setValue(nodeActor, " .. sFirst .. ".starttime, number, " .. nStartTime .. ") = ", DB.setValue(nodeActor, "" .. sFirst .. ".starttime", "number", nStartTime));
 end
```
Comment (JustinFreitas) [inline scripts/manager_time.lua:24]: Cleaned up this function.  The arg coming in is a node, so I eliminated the redundant assignment and extraneous arg to the current date call.  Got rid of the global use too.

File: scripts/manager_time.lua [hunk 2]
```diff
@@ -18,23 +18,13 @@ function onInit()
...
-function getStartTime(rActor, sFirst)
-	-- Debug.console("getStartTime called; " .. sFirst .."");
-	local nodeActor = rActor;
-	FetchStartTime = DB.getValue(nodeActor, "starttime", 0);
-	-- Debug.console("setStartTime; FetchStartTime = DB.getValue(" .. nodeActor .. ", " .. sFirst .. ".starttime, " .. nStartTime .. ") = " .. DB.getValue(nodeActor, "" .. sFirst .. ".starttime", nStartTime) .. "");
-
-	return FetchStartTime;
+function getStartTime(nodeActor)
+	return DB.getValue(nodeActor, "starttime", 0);
 end
```
Comment (JustinFreitas) [inline scripts/manager_time.lua:28]: Cleaned up this function.  Got rid of variables and just return the value.  Also, the input arg is already a node.

File: desktop/desktop_panels.xml [hunk 3]
```diff
@@ -431,6 +439,8 @@
 						local nCurrentRound = DB.getValue(CombatManager.CT_ROUND, 0);
 						nCurrentRound = nCurrentRound % 10
 						DB.setValue(CombatManager.CT_ROUND, 'number', nCurrentRound);
+
+						DB.setValue("calendar.addyear", "number", getValue() or 0);
 					end
 				</script>
 			</basicnumber>
```
Comment (JustinFreitas) [inline desktop/desktop_panels.xml:446]: All of these panel changes do the same things... in the onInit() for the number control, it gets the value from the DB or uses zero if not found.  Then in the onClose() and onDoubleClick/Update() the value of the control is saved in the DB.

File: scripts/manager_time.lua (hunk 1 above)
Comment (bmos) [reply scripts/manager_time.lua:24]: Yeah! I've noticed there are a lot of globals in ClockAdjuster. They don't seem to be fouling anything up yet, but not ideal!

**Pull request merged**